cmd/internal/types.go

```diff
@@ -160,6 +161,8 @@ func parseValue(val sqltypes.Value, columnType string) sqltypes.Value {
 	} else if strings.HasPrefix(columnType, "set") {
 		values := parseEnumOrSetValues(columnType)
 		return mapSetValue(val, values)
```

Comment:
All of the enum and set code above is wrong after https://github.com/vitessio/vitess/pull/15723

That work is in v20+ in OSS and private.

Reply:
I think I'll remove it once everyone is on 20+ 😄 By "wrong" do you mean "unnecessary" or do you mean it will actually return the wrong values?

Reply:
It would be wrong if the user has string values in the ENUM/SET that can be converted to numbers. For example, the string value may be `2` but it's not the index value of 2.

And we can do this work only when necessary by doing it conditional on:
  1. We're NOT in the copy phase
  2. The `EnumSetStringValues` != TRUE
  
  See: https://github.com/vitessio/vitess/blob/main/changelog/20.0/20.0.0/summary.md#enum-set-vstream
  
  That is exactly what they did in the Debezium connector.

Reply:
👍 We will address this in the next PR!